state/runtime/evm/instructions.go

```diff
@@ -813,17 +813,19 @@ func opReturnDataCopy(c *state) {
 	dataOffset := c.pop()
 	length := c.pop()
 
-	if !c.allocateMemory(memOffset, length) {
+	if !dataOffset.IsUint64() {
+		c.exit(errReturnDataOutOfBounds)
+
 		return
 	}
 
-	if !dataOffset.IsUint64() {
-		c.exit(errGasUintOverflow)
-
+	// if length is 0, return immediately since no need for the data copying nor memory allocation
+	if length.Sign() == 0 || !c.allocateMemory(memOffset, length) {
```

Comment:
Not to forget, that maybe we should check is it safe to remove size check from `allocateMemory` function (https://github.com/0xPolygon/polygon-edge/blob/feature/v3-parity/state/runtime/evm/state.go#L306)? Would be good if it is safe to remove it from there.